mne/viz/backends/_pysurfer_mayavi.py

```diff
@@ -65,10 +65,34 @@ def set_interactive():
         tvtk.InteractorStyleTerrain()
 
 
-def surface(surface, color, opacity=1.0, backface_culling=False):
+def mesh(x, y, z, triangles, color, opacity=1.0, shading=False,
+         backface_culling=False, **kwargs):
+    surface = renderer.mlab.triangular_mesh(x, y, z, color=color,
+                                            opacity=opacity,
+                                            figure=renderer.fig,
+                                            **kwargs)
+    surface.actor.property.shading = shading
+    surface.actor.property.backface_culling = backface_culling
+
+
+def contour(surface, scalars, contours, line_width=1.0, opacity=1.0,
+            vmin=None, vmax=None, colormap=None):
+    mesh = _create_mesh_surf(surface, renderer.fig, scalars=scalars)
+    cont = renderer.mlab.pipeline.contour_surface(
+        mesh, contours=contours, line_width=1.0, vmin=vmin, vmax=vmax,
+        opacity=opacity, figure=renderer.fig)
+    cont.module_manager.scalar_lut_manager.lut.table = colormap
+
+
+def surface(surface, color=(0.7, 0.7, 0.7), opacity=1.0,
```

Comment:
what is the conceptual difference between surface and mesh?

Reply:
I would say surface has a more generic definition, you can define a surface by an equation for example while a mesh will be described by its combinatorial elements : vertice, edge and triangle.